### DISCUSSION TITLE
Visual Studio Code extension proxy

### DISCUSSION THREAD

**Discussion by SvenAelterman** [microsoft/AzureTRE#3787] [Q&A]
Has anyone considered how users can securely get and update Visual Studio Code extensions?

I suppose it's possible to unblock the Marketplace URLs in the Firewall, but just like with other package managers, there is the possibility for a user to exfiltrate data by publishing an extension.

**Comment by marrobi** [comment]
@martinpeck did you come across this recently?

**Reply by martinpeck** [reply]
We've pre-installed extensions as part of a Custom VM, but not opened up rules in the firewall.

**Comment by marrobi** [comment]
Could proxy the URLs through nexus form manual download maybe and then install offline?

https://stackoverflow.com/questions/37071388/how-can-i-install-visual-studio-code-extensions-offline/38866913#38866913

Also see: https://github.com/Microsoft/vscode/issues/21839
